Source/Scene/Cesium3DTileFeature.js

```diff
@@ -0,0 +1,247 @@
+/*global define*/
+define([
+        '../Core/Color',
+        '../Core/defined',
+        '../Core/defineProperties'
+    ], function(
+        Color,
+        defined,
+        defineProperties) {
+    'use strict';
+
+    /**
+     * Provides access to a feature's properties stored in the tile's batch table, as well
+     * as the ability to show/hide a feature and change its highlight color via
+     * {@link Cesium3DTileFeature#show} and {@link Cesium3DTileFeature#color}, respectively.
+     * <p>
+     * Modifications to a <code>Cesium3DTileFeature</code> object have the lifetime of the tile's
+     * content.  If the tile's content is unloaded, e.g., due to it going out of view and needing
+     * to free space in the cache for visible tiles, listen to the {@link Cesium3DTileset#tileUnload} event to save any
+     * modifications. Also listen to the {@link Cesium3DTileset#tileVisible} event to reapply any modifications.
+     * </p>
+     * <p>
+     * Do not construct this directly.  Access it through {@link Cesium3DTileContent#getFeature}
+     * or picking using {@link Scene#pick} and {@link Scene#pickPosition}.
+     * </p>
+     *
+     * @alias Cesium3DTileFeature
+     * @constructor
+     *
+     * @example
+     * // On mouse over, display all the properties for a feature in the console log.
+     * handler.setInputAction(function(movement) {
+     *     var feature = scene.pick(movement.endPosition);
+     *     if (feature instanceof Cesium.Cesium3DTileFeature) {
+     *         var propertyNames = feature.getPropertyNames();
+     *         var length = propertyNames.length;
+     *         for (var i = 0; i < length; ++i) {
+     *             var propertyName = propertyNames[i];
+     *             console.log(propertyName + ': ' + feature.getProperty(propertyName));
+     *         }
+     *     }
+     * }, Cesium.ScreenSpaceEventType.MOUSE_MOVE);
+     */
+    function Cesium3DTileFeature(tileset, content, batchId) {
+        this._content = content;
+        this._batchId = batchId;
+        this._color = undefined;  // for calling getColor
+
+        /**
+         * All objects returned by {@link Scene#pick} have a <code>primitive</code> property.
```

Comment:
This doc could be improved.  It took me a second to realize this was the `tileset` passed into the constructor because the property is called `primitive`.